lib/api/core/storage/storageEngine.js

```diff
@@ -144,28 +150,69 @@ class IndexCache {
   remove ({ index, collection, scope='public', notify=true }) {
     let modified = false;
 
-    if (this.indexes[index] && this.indexes[index].scope === scope) {
+    if (this._indexes[index] && this._indexes[index].scope === scope) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the nested one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ACollapsibleIfStatements)